snc.sh

```diff
@@ -245,6 +241,39 @@ retry ${OC} delete mc chronyd-mask
 # Wait for the cluster again to become stable because of all the patches/changes
 wait_till_cluster_stable
 
+# This section is used to create a custom-os image which have `/Users`
+# For more details check https://github.com/crc-org/snc/issues/1041#issuecomment-2785928976
+# This should be performed before removing pull secret
+# Unsetting KUBECONFIG is required because it has default `system:admin` user which doesn't able to create
+# token to login to registry and kubeadmin user is required for that.
+unset KUBECONFIG
+RHCOS_IMAGE=$(${OC} adm release info -a ${OPENSHIFT_PULL_SECRET_PATH} ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE} --image-for=rhel-coreos)
+cat << EOF > ${INSTALL_DIR}/Containerfile
+FROM scratch
+RUN ln -sf var/Users /Users && mkdir /var/Users
+EOF
+podman build --from ${RHCOS_IMAGE} --authfile ${OPENSHIFT_PULL_SECRET_PATH} -t default-route-openshift-image-registry.apps-crc.testing/openshift-machine-config-operator/rhcos:latest --file ${INSTALL_DIR}/Containerfile .
+retry ${OC} login -u kubeadmin -p $(cat ${INSTALL_DIR}/auth/kubeadmin-password) --insecure-skip-tls-verify=true api.${SNC_PRODUCT_NAME}.${BASE_DOMAIN}:6443
+retry ${OC} registry login -a ${INSTALL_DIR}/reg.json
+podman push --authfile ${INSTALL_DIR}/reg.json --tls-verify=false default-route-openshift-image-registry.apps-crc.testing/openshift-machine-config-operator/rhcos:latest
+cat << EOF > ${INSTALL_DIR}/custom-os-mc.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: custom-image
+spec:
+  osImageURL: image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/rhcos:latest
```

Comment:
how much does the it increase the bundle size? and can we later delete the image after machine config pool is updated?

Reply:
Yes that's what I am exploring.

Reply:
I checked the images if it is stored in bundle after creating the bundle
```
rhel-coreos                                    sha256:66b5f9d8f1af5acee783db84a25c08338252ff9c276d33e391680168a22f42ee
```

but it is not available in the bundle.